common/ferrostar/src/routing_adapters/osrm/models.rs

```diff
@@ -67,34 +69,44 @@ pub struct RouteLeg {
     pub via_waypoints: Vec<ViaWaypoint>,
 }
 
-/// An annotation of a route leg with fine-grained information about segments or nodes.
-#[derive(Deserialize, Debug)]
-pub struct Annotation {
-    /// The duration between each pair of coordinates, in seconds.
-    pub duration: Vec<f64>,
-    /// The distance between each pair of coordinates, in meters.
-    pub distance: Vec<f64>,
-
-    /// The average speed used in the calculation between the two points in each pair of
-    /// coordinates, in meters per second.
-    ///
-    /// NOTE: This annotation is not in the official spec, but is a common extension used by Mapbox
-    /// and Valhalla.
-    #[serde(default)]
-    pub speed: Vec<f64>,
-    /// The local posted speed limit between each pair of coordinates.
-    ///
-    /// NOTE: This annotation is not in the official spec, but is a common extension used by Mapbox
-    /// and Valhalla.
-    #[serde(default, rename = "maxspeed")]
-    #[allow(dead_code)]
-    max_speed: Vec<MaxSpeed>,
+#[derive(Deserialize, Debug, Clone, PartialEq)]
+pub struct AnyAnnotation {
+    #[serde(flatten)]
+    pub values: BTreeMap<String, Vec<Value>>,
+}
+
+/// The max speed json units that can be associated with annotations.
+/// [MaxSpeed/Units](https://wiki.openstreetmap.org/wiki/Map_Features/Units#Speed)
+/// TODO: We could generalize this as or map from this to a `UnitSpeed` enum.
+#[derive(Deserialize, PartialEq, Debug, Clone)]
+pub enum MaxSpeedUnits {
+    #[serde(rename = "km/h")]
+    KilometersPerHour,
+    #[serde(rename = "mph")]
+    MilesPerHour,
 }
 
 /// The local posted speed limit between a pair of coordinates.
-#[derive(Deserialize, Debug)]
-pub struct MaxSpeed {
-    // TODO
+#[derive(Debug, Clone)]
+#[allow(dead_code)] // TODO: https://github.com/stadiamaps/ferrostar/issues/271
+pub enum MaxSpeed {
+    Unknown { unknown: bool },
+    Known { speed: f64, unit: MaxSpeedUnits },
+}
+
+#[allow(dead_code)] // TODO: https://github.com/stadiamaps/ferrostar/issues/271
```

Comment:
I think it's fine to leave the defaults (I think they are just warnings, right?). They remind us that there is still something unfinished.

Also, do you have any thoughts on using `SpeedLimit` instead of `MaxSpeed`? I think it's more descriptive and not so Valhalla-specific.

Unrelated: I thought for sure there was an "unlimited" speed. Maybe that's just internal to Valhalla but they might report autobahns with a different speed than unknown or a max value.

Reply:
Yeah, just warnings, so we could definitely revert and leave the todo. 

As for naming, I think we can table this one until the serialization of this object ticket is sorted. It's entirely reasonable to assume that we'll parse from `"max_speed"` on valhalla, mapbox type routes then convert to our more router flexible `SpeedLimit` object.

Reply:
Ooh! Sorry, I missed that this is an OSRM-specific module. It's the public one that I think _has_ to be more generally named. Though TBH I think naming the struct `SpeedLimit` here too is far more useful to the average reader than `MaxSpeed`. IT's a fine line though. A comment is enough I think.